src/arrayMethodJoin.js

```diff
@@ -5,7 +5,34 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let separatorValue = separator;
```

Comment:
You may use the `separator` parameter and get rid of this declaration along with if, if you add a default value in the parameter declaration
```js
[].__proto__.join2 = function(separator = ',') {
```